examples/light_sense.rs

```diff
@@ -0,0 +1,223 @@
+#![no_std]
+#![no_main]
+#![feature(lang_items, core, no_std)]
+
+extern crate emlib;
+extern crate core;
+
+use core::default::Default;
+use emlib::cmsis::nvic;
+use emlib::{acmp, cmu, gpio, irq, lesense, emu, rtc};
+
+// ACMP
+const ACMP_THRESHOLD: u16 = 0x38;
+
+const USER_LED_SECONDS_ON: u32 = 2;
+
+// RTC
+const RTC_FREQ: u32       = 32768;
+const RTC_COMP_VALUE: u32 = (RTC_FREQ * USER_LED_SECONDS_ON);
+
+// GPIO
+const LED_GPIO_PORT: gpio::Port = gpio::Port::E;
+const LED_GPIO_PIN: u32 = 2;
+// LESENSE
+const LIGHTSENSE_CH: u32          = 6;
+const LIGHTSENSE_EXCITE_PORT: gpio::Port = gpio::Port::D;
+const LIGHTSENSE_EXCITE_PIN: u32  = 6;
+const LIGHTSENSE_SENSOR_PORT: gpio::Port = gpio::Port::C;
+const LIGHTSENSE_SENSOR_PIN: u32  = 6;
+const LCSENSE_SCAN_FREQ: u32      = 20;
+const LIGHTSENSE_INTERRUPT: u32   = (0x1 << 6); // LESENSE_IF_CH6
+
+fn setup_cmu() {
+    cmu::clock_enable(cmu::Clock::ACMP0, true);
+    cmu::clock_enable(cmu::Clock::GPIO, true);
+    /* Low energy peripherals
+    *   LESENSE
+    *   LFRCO clock must be enabled prior to enabling
+    *   clock for the low energy peripherals */
+    cmu::clock_select_set(cmu::Clock::LFA, cmu::Select::LFRCO);
+    cmu::clock_enable(cmu::Clock::CORELE, true);
+    cmu::clock_enable(cmu::Clock::LESENSE, true);
+    /* RTC */
+    cmu::clock_enable(cmu::Clock::RTC, true);
+    /* Disable clock source for LFB clock. */
+    cmu::clock_select_set(cmu::Clock::LFB, cmu::Select::Disabled);
+}
+
+fn setup_acmp() {
+    /* Configuration structure for ACMP */
+    let acmp_init = acmp::Init {
```

Comment:
Same as the other, I wanted to keep the comments for the sake of a complete example. Is that not good?